design/ceph-config-updates.md

```diff
@@ -0,0 +1,235 @@
+proposed herein is to add `osd pool default {,min} size = 1` to the cluster CRD in Rook's
+`cluster/example/kubernetes/ceph` directory so that users "just trying out Rook" still get today's
+easy experience but where production clusters do not by default allow users to run clusters with
+potentially dangerous settings.
+
+### Minimal command line options
+
+The minimal set of command line options is pared down to the settings which inform the daemon where
+to find and/or store data.
+
+Required flags:
+- `--{public,cluster}-addr` are not required for most daemons but ...
+  - `--public-addr` and `--public-bind-addr` are necessary for mons
+  - `--public-addr` and `--cluster-addr` may be needed for osds
+    **TODO** find out for sure ^
+  - it may be more appropriate to set `--{public,cluster}-network` instead of `-addr`
+- `--keyring` may necessary to inform daemons where to find the keyring which must be mounted to a
+  nonstandard directory by virtue of being sourced via a Kubernetes secret.
+  - The keyring could copied from the secret mount location to Ceph's default location with an init
+    container, and for some daemons, this may be necessary. This should not be done for the mons,
+    however, as the mon keyrings also include the admin keyring, and persisting the admin key to
+    disk should be avoided at all costs for security.
+
+Notable non-required flags:
+- `--{mon,osd,mgr,mds,rgw}-data-dir` settings exist for all daemons, but it is more desirable to use
+  the `/var/lib/ceph/<daemon_type>/ceph-<daemon_id>` directory for daemon data within containers. If
+  possible, mapping the `dataDirHostPath/<rook_daemon_data_dir>` path on hosts to this default
+  location in the container is preferred.
+  - Note that currently, `dataDirHostPath` is mapped directly to containers, meaning that each
+    daemon container has access to other daemon containers' host-persisted data. Modifying Rook's
+    behavior to only mount the individual daemon's data dir into the container as proposed here will
+    be a small security improvement on the existing behavior.
+- `--run-dir` exists for all daemons, but it is likewise more desirable to use the `/var/run/ceph`
+  path in containers. Additionally, this directory stores only unix domain sockets, and it does not
+  need to be persisted to the host. We propose to simply use the `/var/run/ceph` location in
+  containers for runtime storage of the data.
+
+### Additional configuration
+
+Additional configuration which Rook sets up initially should be done by setting values in Ceph's
+centrally-stored config. A large group of additional configurations can be configured at once via
+the Ceph command `ceph config assimilage-conf`. Care should be taken to make sure that Rook does not
+modify preexisting user-specified values.
+**TODO** how to determine if a user has specified a value or if it is at default state?
```

Comment:
Since both rook and the user would be configuring via the mon-based config, i don't think we can distinguish between the two.  However, in rook's case, the values are going to be set initially at cluster create time before the user has a chance to do anything, so there'd be no conflict there.  It would only be if the CRD specified foo=bar and at some later point in time the user says foo=baz that the operator might want to avoid switching it back to foo=bar.  Is that the case you're thinking of?

Reply:
Capturing answer from huddle: When values have not been set, they are not listed in Ceph's config dump. However, for the first iteration, Rook should just set the values it wants and not try to be too smart about not overriding user values.